### PR TITLE
[Profile] Add memory statiscs & annotation

### PR DESCRIPTION
**Pull request by jihochu**
It add two functions:
- provide PROFILE_MEM_ANNOTATE macro
- print average and maximum usage of memory.

Signed-off-by: Jiho Chu <jiho.chu@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: nntrainer/utils/profiler.cpp [hunk 1]
```diff
@@ -188,27 +194,33 @@ void GenericProfileListener::report(std::ostream &out) const {
   /// creating header
   // clang-format off
   out << std::setw(column_size[0]) << "event"
-      << std::setw(column_size[1]) << "current size"
-      << std::setw(column_size[2]) << "info"
-      << std::setw(column_size[3]) << "dur" << '\n';
+      << std::setw(column_size[1]) << "size"
+      << std::setw(column_size[2]) << "total"
+      << std::setw(column_size[2]) << ""
```
Comment (songgot) [inline nntrainer/utils/profiler.cpp:199]: is it same with  " << std::setw(column_size[2]) << "total"<<'\n'<<'n' " ?

**Review by songgot — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM. The actual usage of this PR is #1988

**Pull request merged**